redwood-lazylayout-schema/src/main/kotlin/app/cash/redwood/lazylayout/widgets.kt

```diff
@@ -25,33 +25,175 @@ import app.cash.redwood.ui.Margin
 
 @Widget(1)
 public data class LazyList(
+  /**
+   * The layout orientation of the list.
+   */
   @Property(1) val isVertical: Boolean,
+
+  /**
+   * Invoked when the user has scrolled the list, such that the `firstVisibleItemIndex` or the
+   * `lastVisibleItemIndex` has changed. When the user performs a fling, [onViewportChanged] will be
+   * invoked multiple times.
+   *
+   * The `firstVisibleItemIndex` is the index of the first partially visible item within the
+   * viewport. The `lastVisibleItemIndex` is the index of the last partially visible item within the
+   * viewport.
+   */
   @Property(2) val onViewportChanged: (firstVisibleItemIndex: Int, lastVisibleItemIndex: Int) -> Unit,
+
+  /**
+   * The number of un-emitted items before the [items] window.
+   *
+   * @see [items]
+   */
   @Property(3) val itemsBefore: Int,
+
+  /**
+   * The number of un-emitted items after the [items] window.
+   *
+   * @see [items]
+   */
   @Property(4) val itemsAfter: Int,
+
+  /**
+   * Sets whether the list's width will wrap its contents ([Constraint.Wrap]) or match the width of
+   * its parent ([Constraint.Fill]).
+   */
   @Property(5) val width: Constraint,
+
+  /**
+   * Sets whether the list's height will wrap its contents ([Constraint.Wrap]) or match the height
+   * of its parent ([Constraint.Fill]).
+   */
   @Property(6) val height: Constraint,
+
+  /**
+   * Applies margin (space) around the list.
+   */
   @Property(7) val margin: Margin,
+
+  /**
+   * If [isVertical], sets the default horizontal alignment for items in this list. Else, sets the
+   * default vertical alignment for items in this list.
+   */
   @Property(8) val crossAxisAlignment: CrossAxisAlignment,
+
+  /**
+   * The last [ScrollItemIndex] programmatically requested by the user.
+   */
   @Property(9) val scrollItemIndex: ScrollItemIndex,
+
+  /**
+   * A block which describes the content of each placeholder item.
```

Comment:
Don’t just give me generic docs, tell me what the contract is!

Each child in this collection is an element of the placeholder pool
Placeholders are used interchangeably
Policy will decide how large the placeholder pool is

Reply:
This property is better documented in `LazyDsl.kt`. Perhaps I can put an `@see` here to make that clearer.

> Policy will decide how large the placeholder pool is

The placeholder pool size is currently hardcoded to 20. We'll have a policy once https://github.com/cashapp/redwood/issues/1500 is implemented, but in the meantime, we distinctly don't have a policy.